csharp/src/Ice/OutgoingResponseFrame.cs

```diff
@@ -9,83 +9,122 @@
+        /// <param name="encoding">The encoding for the frame payload.</param>
+        public static OutgoingResponseFrame Empty(Encoding encoding)
+            => new OutgoingResponseFrame(encoding, true);
 
-        /// <summary>Creates a new outgoing response frame. It's a partial frame, and its payload needs to be
-        /// written later on with StartReturnValue/EndReturnValue.</summary>
-        /// <param name="current">The current parameter holds decoded header data and other information about the
-        /// request for which this method creates a response.</param>
-        public OutgoingResponseFrame(Current current) : base(current.Adapter.Communicator, Ice1Definitions.Encoding)
+        private OutgoingResponseFrame(Encoding encoding, bool writeEmptyPayload)
         {
-            RequestId = current.RequestId;
-            _payloadEncoding = current.Encoding;
+            Encoding = encoding;
+            Data = new List<ArraySegment<byte>>();
+            if (writeEmptyPayload)
+            {
+                Encoding.CheckSupported();
+                byte[] buffer = new byte[256];
+                int pos = 0;
+                buffer[pos++] = (byte)ReplyStatus.OK;
+                OutputStream.WriteInt(6, buffer.AsSpan(pos, 4));
+                pos += 4;
+                buffer[pos++] = encoding.Major;
+                buffer[pos++] = encoding.Minor;
+                Data.Add(new ArraySegment<byte>(buffer, 0, pos));
+                _payloadEnd = new OutputStream.Position(0, pos);
+                Size = Data.GetByteCount();
+                IsSealed = true;
+            }
+        }
 
-            WriteSpan(Ice1Definitions.ReplyHeader.AsSpan());
-            WriteInt(RequestId);
+        /// <summary>Creates a new outgoing response frame. It's a partial frame, and its payload needs to be
+        /// written later on with WritePayload.</summary>
+        /// <param name="encoding">The encoding for the frame payload.</param>
+        public OutgoingResponseFrame(Encoding encoding) : this(encoding, false)
+        {
         }
 
         /// <summary>Creates a new outgoing response frame with the given reply status and payload.</summary>
-        /// <param name="current">The current parameter holds decoded header data and other information about the
-        /// request for which this method creates a response.</param>
-        /// <param name="replyStatus">The reply status.</param>
+        /// <param name="encoding">The encoding for the frame payload.</param>
         /// <param name="payload">The payload for this response frame.</param>
         // TODO: add parameter such as "bool assumeOwnership" once we add memory pooling.
-        public OutgoingResponseFrame(Current current, ReplyStatus replyStatus, ArraySegment<byte> payload)
-            : this(current)
+        public OutgoingResponseFrame(Encoding encoding, ArraySegment<byte> payload) : this(encoding, false)
```

Comment:
`: this(encoding, writeVoidReturnValue: false)` and it should be on the next line